wolfssl/openssl/pkcs7.h

```diff
@@ -45,6 +48,32 @@ typedef struct WOLFSSL_PKCS7
     int len;
 } WOLFSSL_PKCS7;
 
+typedef struct MimeParameter MimeParameter;
+typedef struct MimeHeader MimeHeader;
+
+struct MimeParameter {
+    MimeParameter*  next;
+    char*           attribute;
+    char*           value;
+};
```

Comment:
Is there a reason not to combine the typedef and struct definition if they are both in pkcs7.h?  See WOLFSSL_PKCS7 as reference.

Reply:
Yes, because the structs contain pointers to their own types, the compiler needs to know what the type is before the struct is defined.